### ISSUE TITLE
Less Compile Error Under BootStrap 2.0.2

### ISSUE THREAD

**Issue by mickdelaney** [andrewdavey/cassette#213]
I've just started a new project using bootstrap 2.0.2 and am getting a less compilation error. 

Rolled back to 2.0.1 and all works fine. - Probably the less compiler version. 

Server Error in '/' Application.

Expected '}' on line 396 in file 'mixins.less':
[395]: .reset-filter() {
[396]:   filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
      --^

Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: Cassette.Stylesheets.LessCompileException: Expected '}' on line 396 in file 'mixins.less':
[395]: .reset-filter() {
[396]:   filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
      --^

Source Error: 

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace: 

[LessCompileException: Expected '}' on line 396 in file 'mixins.less':
[395]: .reset-filter() {
[396]:   filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
       --^

   Cassette.Stylesheets.LessCompiler.Compile(String source, IFile sourceFile) +250
   Cassette.BundleProcessing.<>c__DisplayClass1.<Transform>b__0() +104
   Cassette.Stylesheets.ExpandCssUrlsAssetTransformer.ReadCss(Func`1 openSourceStream) +53
   Cassette.Stylesheets.<>c__DisplayClass1.<Transform>b__0() +43
   Cassette.AssetBase.OpenStream() +103
   Cassette.BundleProcessing.ConcatenatedStreamBuilder.Visit(IAsset asset) +39
   Cassette.FileAsset.Accept(IBundleVisitor visitor) +10
   Cassette.Bundle.Accept(IBundleVisitor visitor) +119
   Cassette.BundleProcessing.AssignHash.Process(Bundle bundle, CassetteSettings settings) +86
   Cassette.BundleProcessing.MutablePipeline`1.Process(T bundle, CassetteSettings settings) +180
   Cassette.Stylesheets.StylesheetBundle.ProcessCore(CassetteSettings settings) +17
   Cassette.Bundle.Process(CassetteSettings settings) +21
   Cassette.Configuration.BundleContainerFactoryBase.ProcessAllBundles(IEnumerable`1 bundles) +205
   Cassette.Configuration.BundleContainerFactory.CreateBundleContainer() +18
   Cassette.CassetteApplicationContainerFactoryBase`1.CreateApplication() +116
   Cassette.CassetteApplicationContainer`1.CreateApplication() +67
   System.Lazy`1.CreateValue() +361
   System.Lazy`1.LazyInitValue() +104
   System.Lazy`1.get_Value() +89
   Cassette.Web.StartUp.InitializeApplicationContainer() +110
   Cassette.Web.StartUp.PostApplicationStart() +217

**Comment by mickdelaney** [comment]
OK, basically bootstrap 2.0.2 uses a very new LESS feature from version 1.3, the dotless guys have added support in the latest code. 

https://github.com/dotless/dotless/issues/155

let me know if u need a hand.... er... and what to do :-)

**Comment by andrewdavey** [comment]
I'm working on v2 Cassette this weekend. I'll update the dotless source to the latest version.

**Comment by worldspawn** [comment]
I see dotless 1.3 is out now on nuget. I can't see a dependency btw cassette and dotless nuget packages - how is that happening?

I am in the process of reskinning a site (with bs 2.0.2) and was hoping to drop the telerik script manager in place of cassette.net!

**Comment by alonando** [comment]
Would it be possible to have DotLess be separate of Cassette so that it can be upgraded independently?

**Comment by andrewdavey** [comment]
I've been speaking with the dotless developers. My fork that separates out the compiler from the asp.net stuff has now been merged into their master branch. So once this makes it into nuget I can make Cassette depend on it (rather than ilrepacking the custom dotless library like I do right now).

**Comment by Stephanvs** [comment]
Any progress on this? I actually ran into the same issue, and wondering when a possible fix will be available. Thanks

**Comment by andrewdavey** [comment]
I'm waiting on this:
https://github.com/dotless/dotless/pull/159#issuecomment-5063852

**Comment by worldspawn** [comment]
hows this travelling? :)

**Comment by hfjallemark** [comment]
Also interested in hearing how this is going:-)

**Comment by andrewdavey** [comment]
There is now a separate package called DotLessClientOnly that contains only the dotless compiler (no asp.net stuff). Cassette [v2](https://github.com/andrewdavey/cassette/tree/v2) is now using this package.

The DotLessClientOnly package has the latest version of the dotless compiler which does work with Twitter Bootstrap.

I'm still putting the finishing touches on Cassette v2 - beta coming very soon. You can of course pull down the latest from the branch, but be ready for currently undocumented breaking changes to the configuration API.

**Comment by hfjallemark** [comment]
Cool stuff Andrew, will check it out.

**Comment by worldspawn** [comment]
Any eta on v2 or more particularly a nuget package that will work with bootstrap 2.0.2?

**Comment by andrewdavey** [comment]
The Cassette v2 beta package has this.
